### PR TITLE
Feat json queries

### PR DESCRIPTION
**Pull request by loks0n**

### CODE REVIEW

File: src/SDK/Language/Web.php [hunk 1]
```diff
@@ -185,6 +185,9 @@ public function getTypeName(array $parameter, array $method = []): string
         if (!empty($parameter['enumValues'])) {
             return \ucfirst($parameter['name']);
         }
+        if (!empty($parameter['name']) && $parameter['name'] === 'queries') {
+            return 'Query[]';
+        }
```
Comment (abnegate) [inline src/SDK/Language/Web.php:190]: I think we might need something like this for Android and Flutter too @lohanidamodar, is there a way to do it without coupling to the parameter name?

File: templates/php/src/Query.php.twig [hunk 2]
```diff
@@ -2,112 +2,145 @@
...
+
+    public function jsonSerialize()
+    {
+        return array_filter([
+            'method' => $this->method,
+            'attribute' => $this->attribute,
+            'values' => $this->values,
+        ], function ($value) {
+            return !is_null($value);
+        });
```
Comment (abnegate) [inline templates/php/src/Query.php.twig:37]: You can omit the callback here, the default is to remove null values

File: templates/ruby/lib/container/query.rb.twig [hunk 3]
```diff
@@ -1,94 +1,127 @@
+require 'json'
+
 module {{spec.title | caseUcfirst}}
     class Query
+        def initialize(method, attribute = nil, values = nil)
+            @method = method
+            @attribute = attribute
+
+            if (values != nil)
```
Comment (abnegate) [inline templates/ruby/lib/container/query.rb.twig:9]: Let's omit braces for if conditions

File: templates/ruby/lib/container/query.rb.twig [hunk 4]
```diff
@@ -1,94 +1,127 @@
...
+
+            if (@attribute != nil)
+                query["attribute"] = @attribute
+            end
+
+            if (@values != nil)
+                query["values"] = @values
+            end
+
+            return query.to_json(*args)
```
Comment (abnegate) [inline templates/ruby/lib/container/query.rb.twig:31]:
```suggestion
            {
                method: @method,
                attribute: @attribute,
                values: @values
            }.compact.to_json(*args)
```

File: templates/dotnet/src/Appwrite/Query.cs.twig [hunk 5]
```diff
@@ -1,139 +1,145 @@
...
 
 namespace Appwrite
 {
-    public static class Query
+    public class Query
     {
-        public static string Equal(string attribute, object value)
+        public string method;
+        public string? attribute;
+        public object? values;
```
Comment (abnegate) [inline templates/dotnet/src/Appwrite/Query.cs.twig:13]: We can use `List<object?>` here

File: templates/swift/Sources/Query.swift.twig [hunk 6]
```diff
@@ -1,104 +1,272 @@
...
+
     }
 
-    public static func isNull(_ attribute: String) -> String {
-        "isNull(\"\(attribute)\")"
+    public static func equal(_ attribute: String, value: Any) -> Query {
+        return Query(
+            method: "equal",
+            attribute: attribute,
+            values: [value]
```
Comment (abnegate) [inline templates/swift/Sources/Query.swift.twig:96]: We need to make sure `value` isn't already an array for methods that accept `Any`

File: templates/web/src/client.ts.twig [hunk 7]
```diff
@@ -359,6 +359,9 @@ class Client {
         } else {
             switch (headers['content-type']) {
                 case 'application/json':
+                    if ('queries' in params && Array.isArray(params['queries'])) {
+                        params['queries'] = params['queries'].map((query) => query.toString());
+                    }
```
Comment (abnegate) [inline templates/web/src/client.ts.twig:364]: We need to do this in the GET case, routes with queries will always be GET

File: templates/deno/src/client.ts.twig [hunk 8]
```diff
@@ -85,6 +85,9 @@ export class Client {
 
             body = formData;
         } else {
+            if ('queries' in params && Array.isArray(params['queries'])) {
+                params['queries'] = params['queries'].map((query) => query.toString());
+            }
```
Comment (abnegate) [inline templates/deno/src/client.ts.twig:90]: We need to do this in the GET case, routes with queries will always be GET

File: src/SDK/Language/Web.php (hunk 1 above)
Comment (loks0n) [reply src/SDK/Language/Web.php:190]: We need to put something else in the specs. This was my workaround for now

File: templates/ruby/lib/container/query.rb.twig (hunk 3 above)
Comment (loks0n) [reply templates/ruby/lib/container/query.rb.twig]: done

File: templates/php/src/Query.php.twig (hunk 2 above)
Comment (loks0n) [reply templates/php/src/Query.php.twig:37]: `json_encode` does not remove the null values. Are you referring to something else? I think it's safer to keep this

File: templates/ruby/lib/container/query.rb.twig (hunk 4 above)
Comment (loks0n) [reply templates/ruby/lib/container/query.rb.twig]: done

File: templates/web/src/client.ts.twig (hunk 7 above)
Comment (loks0n) [reply templates/web/src/client.ts.twig]: Done inside flatten() function instead.

File: templates/deno/src/client.ts.twig (hunk 8 above)
Comment (loks0n) [reply templates/deno/src/client.ts.twig]: Done inside flatten() function instead.

File: templates/dotnet/src/Appwrite/Query.cs.twig (hunk 5 above)
Comment (loks0n) [reply templates/dotnet/src/Appwrite/Query.cs.twig]: done

File: templates/php/src/Query.php.twig (hunk 2 above)
Comment (abnegate) [reply templates/php/src/Query.php.twig:37]: From the `array_filter`, we can just have it like:

```suggestion
        ]);
```

**Review by abnegate — APPROVED** [review]

**Pull request merged**